flowmachine/flowmachine/features/subscriber/rolling_count_threshold_subscribers.py

```diff
@@ -0,0 +1,61 @@
+from functools import reduce
+from typing import List
+from dateutil.rrule import DAILY, rrule
+from dateutil.relativedelta import relativedelta
+
+from flowmachine.core.mixins.exposed_datetime_mixin import ExposedDatetimeMixin
+from flowmachine.core.query import Query
+from flowmachine.features.subscriber.active_subscribers import ActiveSubscribers
+
+
+class RollingCountThresholdSubscribers(ExposedDatetimeMixin, Query):
```

Comment:
I'm questioning whether this query in this form is actually necessary. I guess this comes back to the question of function vs purpose. We could have a query that just takes a list of `ActiveSubscribers` queries, and returns the set of unique subscribers from the union of those queries - that would be functionally equivalent to this, but more general-purpose in that a user could pass in any list of queries (such as active subscribers during each of three disconnected periods, rather than a rolling window).

If the process of defining the rolling-window set of `ActiveSubscriber` queries is too much of a burden to leave for the user to implement outside of FlowKit (by which I essentially mean "if the rolling-window set of `ActiveSubscriber` queries is something we'll be using regularly"), I'd argue this functionality would be better-placed in a FlowClient helper function (similar to the `modal_location_from_dates` flowclient function, as a convenience alternative to `modal_location`).

Reply:
Assuming the `ActiveSubscriber` queries have only one column, `union` with `all=False` will actually do the whole thing really.

Reply:
That's true. We'd still need to make that functionality accessible through the API in some way.

I wonder whether we could push this functionality down into the subscriber subsetter (so that the `subscriber_subset` parameter could take a list of queries, and the unioning and uniqueification is handled by the subsetter query)?

Reply:
Links to #1009 on the api exposure.

We definitely _could_ pass it down, but whether one should is maybe another question. What if you actually _want_ to sample-with-replacement, or something?

Reply:
True. It seems to me the simplest option would be to add a new `UniqueSubscribersFromQueries` query which basically just wraps the `union(all=False)` functionality (we could make handle queries with additional columns as well, which would make it feel a bit less redundant). Then we can expose that as an input to the `subscriber_subset` parameter through the API. How does that sound?

Reply:
That sounds reasonable to me. I don't think one is compelled to use the union class here btw - would also seem totally reasonable to do it in sql (probably better, because you don't then have a heavily nested union, and can do a select subscriber from ... on the input queries).